codegen/value.go

```diff
@@ -55,8 +55,14 @@ func NewRegister(ctx context.Context) Register {
 }
 
 func (r *register) Set(iface interface{}) error {
+	val, err := r.ctor(iface)
+	if r.last == nil {
+		// If there are no async queued up, fast path towards setting the register.
+		r.value = val
```

Comment:
Is it correct to set `r.value = val` even if `r.ctor` returned an error? I notice the slow path would set it to `&errorValue{err}`.

Reply:
Yes you're right, we don't need to set `&errorValue{err}` but we shouldn't override `r.value` with a nil.